test-data/unit/fine-grained-suggest.test

```diff
@@ -1090,7 +1090,7 @@ optional2(10)
 optional2('test')
 
 def optional3(x: Optional[List[Any]]):
-    assert not x
+    assert x
```

Comment:
I didn't quite understand this one!

Reply:
Tbh I didn't either. `assert not x` doesn't make much sense here. I only excludes the non-empty list. With strict optional, that results in an error for the next line where
```py
    return x[0]  # E: Value of type "Optional[List[Any]]" is not indexable
```

To keep the intention, the assert should probably be `assert not x and x is not None` but I wasn't sure this would be better than just `assert x`.